### PR TITLE
react-command-convert-to-pdf - Updated

### PR DESCRIPTION
**Pull request by NicolasKheirallah**
> By submitting this pull request, you agree to the [contribution guidelines](https://github.com/pnp/sp-dev-fx-extensions/blob/main/CONTRIBUTING.md)

> If you aren't familiar with how to contribute to open-source repositories using GitHub, or if you find the instructions on this page confusing, [sign up](https://forms.office.com/Pages/ResponsePage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUREZVRDVYUUJLT1VNRDM4SjhGMlpUNzBORy4u) for one of our [Sharing is Caring](https://pnp.github.io/sharing-is-caring/#pnp-sic-events) events. It's completely free, and we'll guide you through the process.

|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                               |
| New feature?    | yes                              |
| New sample?     | no                              |
| Related issues? | Updated to latest SPFX, Rewrote and cleaned up alot and moved from using third party to using Graph to both get request the file but also converting it! |

## What's in this Pull Request?

> Please describe the changes in this PR. Sample description or details around bugs which are being fixed.
> Upgrade SPFx to version 1.20.1, rewrite it from scratch using Graph API and using the built in converter in Graph, moved to Fluent UI and alot more

### CODE REVIEW

File: samples/react-command-convert-to-pdf/gulpfile.js [hunk 1]
```diff
@@ -34,4 +42,7 @@ build.rig.getTasks = function () {
   return result;
 };
 
-build.initialize(gulp);
\ No newline at end of file
+
+require("./spfx-versioning")(build);
+build.tslintCmd.enabled = false;
```
Comment (Adam-it) [inline samples/react-command-convert-to-pdf/gulpfile.js:47]: this is not a huge problem or a bug but seems to me a bit out of scope for the related issue. Our aim is to upgrade the project and fix it up if anything requires updates to make it work with SPFx 1.20. 
Here we are adding an additional file for auto update of version number. I say lets avoid including additional side changes and make our update and PR simple and clean 👍

**Review by Adam-it — CHANGES_REQUESTED** [review]
@NicolasKheirallah Awesome work so far 👏👏👏👏
I think we are on the right track but it seems w have a few tiny steps w need to sort out before we proceed. I left a few comments, please do give the a double check 🙏

Also we should update the `assets/sample.json` file. We should update the `updateDateTime` and `SPFX-VERSION` metadata

Also did you generate the upgrade guidance for this project and go over the required things? 
I noticed some were still not done like including `eslint` or modifying the `yo-rc.json`
image

File: samples/react-command-convert-to-pdf/gulpfile.js (hunk 1 above)
Comment (NicolasKheirallah) [reply samples/react-command-convert-to-pdf/gulpfile.js:47]: Thanks for comments, feedback and guidance!
I've never used the SPFX upgrade CLI, Didn't know it existed!! That's gonna help quite a bit!

I'll fix it right away and make a new pull request! :)

Comment (Adam-it) [reply samples/react-command-convert-to-pdf/gulpfile.js:47]: no problem. in the related issue I mentioned two tools that might support you in the upgrade. 
You may use either CLI for Microsoft 365 using the `m365 spfx project upgrade` command 
or You may install the SPFx Toolkit VS Code extension and there is an upgrade action as well

**Comment by Adam-it** [comment]
@NicolasKheirallah are you still working on this? If the PR is ready for review please click on the 'Ready for review' button to let us know

**Comment by NicolasKheirallah** [comment]
> @NicolasKheirallah are you still working on this? If the PR is ready for review please click on the 'Ready for review' button to let us know

I must been blind, didn't even see that button 🤣

File: samples/react-command-convert-to-pdf/README.md
[the remark this answers is not in the document]
Comment (Adam-it) [reply samples/react-command-convert-to-pdf/README.md:1]: I think we are still missing this update. When I look at the readme file we still did not udpate the SPFx version and node version tags
image

Comment (Adam-it) [inline samples/react-command-convert-to-pdf/README.md:1]: We should update the SPFx version and Node version tags

File: samples/react-command-convert-to-pdf/src/extensions/pdfExport/PdfExportCommandSet.tsx
Comment (Adam-it) [inline samples/react-command-convert-to-pdf/src/extensions/pdfExport/PdfExportCommandSet.tsx:1]: why change this file to `tsx`? I think it is not rendering anything right?

**Review by Adam-it — CHANGES_REQUESTED** [review]
@NicolasKheirallah I think we are almost there. I still noticed some things we missed or which is strange. Could you have a closer look?

File: samples/react-command-convert-to-pdf/src/extensions/pdfExport/PdfExportCommandSet.tsx
Comment (NicolasKheirallah) [reply samples/react-command-convert-to-pdf/src/extensions/pdfExport/PdfExportCommandSet.tsx:1]: Had some rendering inside it when I refactored it and trying some stuff and I forgot to change it back 😅

File: samples/react-command-convert-to-pdf/README.md
Comment (NicolasKheirallah) [reply samples/react-command-convert-to-pdf/README.md:1]: Yes! :) Thanks so much for the help!!

**Review by Adam-it — APPROVED** [review]
@NicolasKheirallah all good 👏👏👏👏

**Pull request merged**